### DISCUSSION TITLE
Support for writing OpenEXR files

### DISCUSSION THREAD

**Discussion by aferrero2707** [libvips/libvips#2025] [Ideas]
VIPS has already very good support for loading OpenEXR files. Would it be possible to also implement writing to OpenEXR file format? That would be extremely useful in all workflows that involve loading/saving/manipulating scene-referred images.

In PhotoFlow I am actually falling back to 32-bit floating-point TIFF files, however the OpenEXR would be more storage efficient and also more commonly used for saving and exchanging HDR content.

Thanks!

**Comment by jcupitt** [comment]
I agree, openexr support needs improving.

As well as openexrsave, I think it should switch to using the C++ API. openexr has a C API, but it's a pale shadow of the C++ one and is missing features like profile support. Now that we support libvips operations in C++, we should use the C++ API.

**Comment by aferrero2707** [comment]
That sounds really great, thanks a lot!

AFAIK the OpenEXR container allows to specify the RGB primaries, but down not allow to embed a full ICC profile. However, I can provide code to create the ICC profile from the primaries, if needed...
